jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -413,30 +474,47 @@ def _preset_args_changed(self, msg={}):
             # https://github.com/spacetelescope/jwst_novt/blob/main/jwst_novt/interact/display.py
             new_marks = []
             for i, reg in enumerate(regs):
+                if (not isinstance(reg, regions.Region)
+                        and not hasattr(reg, 'to_pixel')):  # pragma: nocover
+                    # NOTE: this is pre-checked for API/file selection in the file-parser
+                    # and built-in presets should be designed to never hit this error
+                    # in the future we may support pixel regions as well, but need to decide how
+                    # to properly handle those scenarios for both WCS and pixel-linking
+                    raise NotImplementedError("regions must all be SkyRegions")
+
                 pixel_region = reg.to_pixel(wcs)
-                if not isinstance(reg, regions.PolygonSkyRegion):  # pragma: nocover
-                    # if we ever want to support plotting centers as well,
-                    # see jwst_novt/interact/display.py

```

Comment:
Do you want to add this comment back in since you didn't include my code with the single pixel region thingy?

Reply:
I don't think we need it anymore.  When the code was generalized for all instruments, I stripped the centers out of that logic as well, and I think the if-elif-else block below (and what you had in your PR) is the obvious place to put the display logic if we ever want to re-introduce that.